### ISSUE TITLE
Further pretrain in domain specific corpus

### ISSUE THREAD

**Issue by hexiaoyupku** [zihangdai/xlnet#230]
Hi, 
I tried to further pretrain XLNet in domain specific corpus like what is recommanded in BERT. But I got worse result. Has anyone tried further pretraining? Does it work?

Thanks!

**Comment by huhk-sysu** [comment]
Hi, I'm tring to do further pretraining with XLNet too. I haved done the same thing with BERT before and confirm its effectiveness, so I wonder why you got worse result on XLNet.To help me avoid the traps, could you explain the following in detail?

1. Which XLNet model did you use?
2. What kind of corpus did you use? How many sentences are there?
3. What are the hyperparameters during pretraining?
4. How does the model behave worse? Like, a drop on accuracy?

I'm not sure whether you have solved the problem, if so, I'm willing to hear your advises.
Thank you!

**Comment by chan4cc** [comment]
Hi, I also got worse result while doing further pre-train on the new corpus.
The task is running on V100(32G) and the seq_len is 256, script for data precess:
```
SAVE_DIR=/data1/####
INPUT=/data1/###/*.txt
num_task=200

for task_num in `seq 1 200`;
do
python3 data_utils.py \
	--bsz_per_host=32 \
	--num_core_per_host=1 \
	--seq_len=256 \
	--reuse_len=128 \
	--input_glob=${INPUT} \
	--save_dir=${SAVE_DIR} \
	--num_passes=20 \
	--bi_data=True \
	--sp_path=spiece.model \
	--mask_alpha=6 \
	--mask_beta=1 \
	--num_predict=40 \
	--uncased=False \
	--num_task=$num_task \
	--task=$[$task_num-1] &
done
```
the pre-train script like this:
```
DATA=/data/#####/tfrecords
MODEL_DIR=/data/######
INIT_CKPT=/data/###/model/chinese_xlnet_base_L-12_H-768_A-12/xlnet_model.ckpt

python3 train_gpu.py \
	--record_info_dir=$DATA \
	--model_dir=$MODEL_DIR \
	--init_checkpoint=$INIT_CKPT \
	--num_core_per_host=1 \
	--train_batch_size=32 \
	--seq_len=256 \
	--reuse_len=128 \
	--mem_len=64 \
	--perm_size=128 \
	--n_layer=12 \
	--d_model=768 \
	--d_embed=768 \
	--n_head=12 \
	--d_head=64 \
	--d_inner=3072 \
	--untie_r=True \
	--mask_alpha=6 \
	--mask_beta=1 \
	--num_predict=40 \
	--uncased=False \
	--train_steps=1500000 \
	--save_steps=30000 \
	--iterations=1000 \
	--warmup_steps=2000 \
	--max_save=20 \
	--weight_decay=0.01 \
	--adam_epsilon=1e-6 \
	--learning_rate=1e-4 \
	--dropout=0.1 \
	--dropatt=0.1
```
Also, I tried to decrease the learning_rate from 1e-4 to 1e-7. 
Is my task configuration or any hyper-parameters wrong，any advice is appreciated，thx!
